src/compiler/checker.ts

```diff
@@ -5098,6 +5098,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (name === InternalSymbolName.ExportEquals) return;
             merged.exports!.set(name, merged.exports!.has(name) ? mergeSymbol(merged.exports!.get(name)!, s) : s);
         });
+        if (merged === exported) {
+            // We just mutated a symbol, reset any cached links we may have already set
+            // (Notably required to make late bound members appear)
+            getSymbolLinks(merged).resolvedExports = undefined;

```

Comment:
There's no precedent for doing this as far as I can tell; is it likely this information has already been read once before and so some places will have stale info?

Reply:
There's also no other place where we mutate the `.exports` of a transient symbol like we do on the lines above, so we just kinda have to. The alternative is what the prior PR does, which is attempt to guarantee an ordering of the operations, but that creates an unsolvable circular dependency that, when caught, breaks a bunch. So this is probably as good as it gets, since whatever stale-cache-data issues this introduces (which I don't think should be _too_ much since we do this export= cjs merge pretty early) only applies to cjs assigned export= objects with late bound members merged into them, without affecting the validity of a bunch of code that already works.